c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,75 @@
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        public static JObject NumericDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            string diff = JsonDiffBetweenTwo(left, right);
+            if(diff == null)
             {
-                if (firstKeys.Contains(key))
-                {
-                    matches.Add(key);
-                }
+                return JObject.Parse(@"{}");
             }
-            foreach(string key in matches)
+            JObject diffJsonObject = JObject.Parse(diff);
+            JObject diffReturnObject = CalculateNumericDiff(diffJsonObject);
+            return diffReturnObject;
+        }
+
+        private static JObject CalculateNumericDiff(JObject jObject)
+        {
+            foreach (KeyValuePair<string, JToken?> prop in jObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if (prop.Value != null)
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    if (!(prop.Value is JArray))
+                    {
+                        CalculateNumericDiff((JObject)prop.Value);
+                    }
+                    else
+                    {
+                        JArray valuesAsArray = (JArray)prop.Value;
+                        if (valuesAsArray[0] != null && valuesAsArray[1] != null)
+                        {
+                            string leftValueAsString = valuesAsArray[0].ToString();
+                            string rightValueAsString = valuesAsArray[1].ToString();
+                            if (int.TryParse(leftValueAsString, out int leftValueAsInt) && int.TryParse(rightValueAsString, out int rightValueAsInt))
+                            {
+                                int diffValue = Math.Abs(leftValueAsInt - rightValueAsInt);
```

Comment:
I don't know if you should be doing `Math.Abs` on the diff. There is a difference between a value increasing or decreasing - the client of this library might need that information.

Maybe add an option to specify whether the diff should be absolute or not? If set to false, just do `left - right`.